managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1794,7 +1794,12 @@ void asyncReadEntries(OpReadEntry opReadEntry) {
     }
 
     public CompletableFuture<String> getLedgerMetadata(long ledgerId) {
-        return getLedgerHandle(ledgerId).thenApply(rh -> rh.getLedgerMetadata().toSafeString());
+        LedgerHandle currentLedger = this.currentLedger;
+        if (currentLedger != null && ledgerId == currentLedger.getId()) {
+            return CompletableFuture.completedFuture(currentLedger.getLedgerMetadata().toSafeString());
```

Comment:
I'm not sure if `currentLedger.getLedgerMetadata().toSafeString()` will throw exception to break the the encapsulation of the method. Because we return the `CompletableFuture` but have a risk throw exception out of the `CompletableFuture` chain.

Could you help to confirm it?

Reply:
I checked this method, it looks safe.